### DISCUSSION TITLE
How should the named colors string interface respond to invalid color names?

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#525] [Ideas]
I got used to teaching "Processing-like" colors, with the `fill(r, g, b)` style color control, in my classes because, well, I've been doing like this for many years... old dog, new tricks, etc.

But hey, I'm thinking about embracing the great *matplotlib* & "CSS4" style string names now! So, today, playing a bit with the strings I decided to try a wrong name to see what happened, and I was a bit surprised by the signature & error shown. 

![image](https://github.com/user-attachments/assets/376fe571-0a92-41d2-a6ac-7a84e9037160)

Do you think this could be improved? I'm not really sure what I expected, but I think I didn't expect  `str` to be flagged as an invalid argument type. Maybe this could be a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) ?

**Comment by hx2A** [comment]
Yes, this can be improved. I agree, that error is confusing. If I mistype a color name, I shouldn't get that exception.

There is `py5.css4_colors` and `py5.xkcd_colors`, which were created because I can't remember the names of more than 5 colors. These should generate a relevant exception if I write `py5.css4_colors.nonexistent_color`. Also, `py5.fill('nonexistent_color')` should not give that TypeError, as found above. If you pass a string and it can't find it, it should throw a different exception. Perhaps something like `"it looks like you are trying to use a named color but 'nonexistent_color' is not a known color"`.

I don't think either of these things are difficult changes to make.

**Reply by hx2A** [reply]
@villares I just opened an issue. This will be an easy fix.
